packages/cozy-doctypes/src/banking/BankingReconciliator.js

```diff
@@ -47,9 +45,14 @@ class BankingReconciliator {
       }
     })
 
-    const stackAccountsIds = new Set(stackAccounts.map(x => x._id))
+    const reconciliatedAccountIds = new Set(
+      reconciliatedAccounts.filter(acc => acc._id).map(acc => acc._id)
+    )
+
+    // Pass to transaction reconciliation only transactions that belong
+    // to one of the reconciliated accounts
     const stackTransactions = (await BankTransaction.fetchAll()).filter(
-      transaction => stackAccountsIds.has(transaction.account)
+      transaction => reconciliatedAccountIds.has(transaction.account)
```

Comment:
We are careful to send only reconciliated accounts to the matching part, since we do not want operations from a different bank to end up there.